src/LibraryManager.Vsix/UI/Controls/Library.xaml.cs

```diff
@@ -14,22 +14,50 @@ namespace Microsoft.Web.LibraryManager.Vsix.UI.Controls
     public partial class Library : INotifyPropertyChanged
     {
         public static readonly DependencyProperty CaretIndexProperty = DependencyProperty.Register(
-            nameof(CaretIndex), typeof(int), typeof(Library), new PropertyMetadata(default(int), SearchCriteriaChanged));
+            nameof(CaretIndex), typeof(int), typeof(Library), new PropertyMetadata(default(int)));
 
         public static readonly DependencyProperty SearchServiceProperty = DependencyProperty.Register(
-            nameof(SearchService), typeof(Func<string, int, Task<CompletionSet>>), typeof(Library), new PropertyMetadata(default(Func<string, int, Task<CompletionSet>>), SearchCriteriaChanged));
+            nameof(SearchService), typeof(Func<string, int, Task<CompletionSet>>), typeof(Library), new PropertyMetadata(default(Func<string, int, Task<CompletionSet>>)));
 
         public static readonly DependencyProperty SelectedItemProperty = DependencyProperty.Register(
             nameof(SelectedItem), typeof(Completion), typeof(Library), new PropertyMetadata(default(Completion)));
 
         public static readonly DependencyProperty TextProperty = DependencyProperty.Register(
-            nameof(Text), typeof(string), typeof(Library), new PropertyMetadata(default(string), SearchCriteriaChanged));
+            nameof(Text), typeof(string), typeof(Library), new PropertyMetadata(default(string)));
 
         private int _version;
 
         public Library()
         {
             InitializeComponent();
+
+            this.Loaded += new RoutedEventHandler(LibrarySearchBox_Loaded);
+        }
+
+        private void LibrarySearchBox_Loaded(object sender, RoutedEventArgs e)
+        {
+            Window window = Window.GetWindow(LibrarySearchBox);
+
+            // Simple hack to make the popup dock to the textbox, so that the popup will be repositioned whenever
+            // the dialog is dragged or resized.
+            // In the below section, we will bump up the HorizontalOffset property of the popup whenever the dialog window
+            // location is changed or window is resized so that the popup gets repositioned.
+            if (window != null)
+            {
+                double offset = Flyout.HorizontalOffset;
+
+                window.LocationChanged += delegate (object s, EventArgs args)
```

Comment:
>delegate (object s, EventArgs args) [](start = 42, length = 35)

Might be cleaner to defined a single method in this class and add it to both events